% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/DebugFn.R
\name{DebugPrintFn}
\alias{DebugPrintFn}
\title{Capture arguments of exception throwing function call for later debugging.}
\usage{
DebugPrintFn(saveFile, fn, ...)
}
\arguments{
\item{saveFile}{path to save RDS to.}

\item{fn}{function to call}

\item{...}{arguments for fn}
}
\value{
fn(...) normally, but if fn(...) throws an exception save to saveFile RDS of list r such that do.call(r$fn,r$args) repeats the call to fn with args.
}
\description{
Run fn and print result, save arguments on failure.  Use on systems like ggplot()
where some calculation is delayed until print().
}
\examples{

saveName <- paste0(tempfile('debug'),'.RDS')
f <- function(i) { (1:10)[[i]] }
# correct run
DebugPrintFn(saveName, f, 5)
# now re-run
# capture error on incorrect run
tryCatch(
   DebugPrintFn(saveName, f, 12),
   error = function(e) { print(e) })
# examine details
situation <- readRDS(saveName)
str(situation)
# fix and re-run
situation$args[[1]] <- 6
do.call(situation$fn,situation$args)
# clean up
file.remove(saveName)

}
\seealso{
\code{\link{DebugFn}}, \code{\link{DebugFnW}},  \code{\link{DebugFnWE}}, \code{\link{DebugPrintFn}}, \code{\link{DebugFnE}}, \code{\link{DebugPrintFnE}}
}

